common-lib/src/main/java/com/google/gct/idea/DefaultPluginConfigurationService.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2015 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gct.idea;
+
+import com.google.gct.idea.feedback.GoogleFeedbackErrorReporter;
+
+import com.intellij.ExtensionPoints;
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginManager;
+import com.intellij.openapi.extensions.ExtensionPointName;
+import com.intellij.openapi.extensions.Extensions;
+import com.intellij.openapi.extensions.PluginId;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This will be the implementation you use for typical plugin deployments to the IntelliJ platform.
+ */
+public class DefaultPluginConfigurationService implements PluginConfigurationService {
+
+  @Override
+  public <T> void registerExtension(
+      @NotNull ExtensionPointName<T> extensionPoint,
+      @NotNull T extension) {
+    Extensions.getRootArea().getExtensionPoint(extensionPoint)
+        .registerExtension(extension);
+  }
+
+  @Override
+  public void enabledGoogleFeedbackErrorReporting(@NotNull String pluginId) {
+    GoogleFeedbackErrorReporter errorReporter = new GoogleFeedbackErrorReporter();
+    IdeaPluginDescriptor plugin = PluginManager.getPlugin(PluginId.getId(pluginId));
+    if (plugin == null) {
+      throw new IllegalArgumentException(pluginId + " is not a valid plugin ID.");
```

Comment:
I think it will have an impact.  I think failures during component initialization causes the plugin to fail to load. 

You do raise a good point that if this fails, we can't capture the fact it fails, and should the plugin be disabled completely at that point?  We certainly would hear about it sooner if we did.